### ISSUE TITLE
Perform security checks on API

### ISSUE THREAD

**Issue by andrea-williams** [bcgov/cas-registration#217]
#### Describe the task

We want to ensure that the authentication mechanisms we have in place to protect our API endpoints are robust.

Therefore, 1-2 people (ideally not the same as the devs who set up API endpoint protection) should stress-test our API to identify any weak spots.

#### Acceptance Criteria

- [ ] check IDIR-protected endpoints
- [ ] check Business BCeID-protected endpoints
- [ ] create additional issues for any fixes required that are surfaced by this testing

#### Additional context

- Add any other context about the task here.
- Or here

Blocked by https://github.com/bcgov/cas-registration/issues/700

**Comment by Dianadec** [comment]
Andrea said we can delay this one to next sprint.

**Comment by andrea-williams** [comment]
As part of this ticket we should also be checking to ensure that external users aren't able to change the status of their operator, operation, or prime admin access request themselves by posting a request using Postman, Thunder Client, or the like (i.e., bypassing the UI). (Although the implications of this will be significantly mitigated by https://github.com/bcgov/cas-registration/issues/503 )

I haven't investigated this yet, but off the top of my head my concern is that we accept values for the "Status" field in the request body on PUT endpoints. We might need to add logic that removes any updates to the "Status" field for operators, operations, or prime admin when the request is coming from an external user.

**Comment by andrea-williams** [comment]
Other places where users should only be allowed to see/edit their own stuff. We'll likely additionally need that protection on:

- @router.get("/is-approved-admin-user-operator/{user_guid}

- @router.get("/operator-has-admin/{operator_id}"

- for all the /operators endpoints, everyone has to be able to see all operators so they can select their operator from the list. However, we currently return the entire operator but we should only be returning the public data (e.g. the stuff we show in the grey box)

**Comment by BCerki** [comment]
It would also be good to confirm that anything that requires approval (e.g. the operator form) is inaccessible for unapproved users

**Comment by andrea-williams** [comment]
Brain-dumping my findings so far (for @Sepehr-Sobhani ) - these are all based on ThunderClient requests to our API:

While logged in as a BCeID with a pending admin request for Operator ID 1:
- [ ] `GET /operators/1` returns all available data about Operator 1. I'm suspicious about this since at this point the user has not been approved 
- [ ] `GET /operators/2` returns all known data about Operator 2. Also suspicious - I shouldn't be able to see much data about other operators
I confirmed that I was not able to send any successful PUT requests to `/operators/1` while my access request was still Pending - I got 401 errors.
- [ ] `POST /user-operator/contact` - I was able to create contact info for Operator 1 when my admin request for that operator was still pending. Furthermore, by changing the value of `user_operator_id` in my request payload, I was able to create new contacts for all operators (whatever operator is associated with the user_operator_id, that's the operator the contact is created for)
- [ ] `PUT /user-profile` - I was able to change my `bceid_business_name` to another business. This could be leveraged to request admin access to operators other than my own. For example: my BCeID account is with Operator 1. I can send a PUT request that modifies my bceid_business_name to Operator 2. Then I make a request to be admin for Operator 2. When CAS internal staff are reviewing my admin request, the UI will say that my bceid account is with Operator 2 (LIES), and will likely then grant me admin access to Operator 2, an operator I don't actually work for.

At the time that I was testing, I was able to POST to `/select-operator/request-admin-access` multiple times so that I had multiple pending admin requests for different operators at the same time, which shouldn't be allowed. However, I think this is about to be fixed in Dylan's latest PR.
